client/dev/sheet/abilitiestable.jsx

```diff
@@ -0,0 +1,122 @@
+import React from "react";
+
+class AbilitiesTable extends React.Component { 
+  constructor( props ){
+    super(props);
+  
+    //Initial state
+    this.state = {
+      data: props.data,
```

Comment:
should not have props inform state; props should be read-only. Instead, have state live in a higher component.